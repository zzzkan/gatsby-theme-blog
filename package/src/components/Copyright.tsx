import React from "react"
import { Text } from "@chakra-ui/react"

type Props = {
  readonly author: string
  readonly publicationYear: number
}

export const Copyright: React.FC<Props> = ({ author, publicationYear }) => {
  return (
    <Text as={"div"} fontSize={"sm"}>
      Â© {publicationYear} {author}
    </Text>
  )
}
