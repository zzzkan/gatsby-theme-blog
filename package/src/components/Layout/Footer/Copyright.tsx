import React from "react"
import { Text } from "@chakra-ui/react"
import { useSiteMetadata } from "../../../hooks/useSiteMetadata"

const Copyright: React.FC = () => {
  const { author } = useSiteMetadata()
  return (
    <Text as={"div"} fontSize={"sm"}>
      Â© 2022 {author}
    </Text>
  )
}

export default Copyright
