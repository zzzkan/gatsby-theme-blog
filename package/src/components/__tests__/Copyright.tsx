import React from "react"
import { render } from "@testing-library/react"
import { Copyright } from "../Copyright"

describe("Copyright component", () => {
  vi.mock("../../hooks/useSiteMetadata", () => {
    return {
      useSiteMetadata: vi.fn().mockImplementation(() => {
        return {
          author: "author",
          publicationYear: 9999,
        }
      }),
    }
  })

  afterEach(() => {
    vi.clearAllMocks()
  })

  afterAll(() => {
    vi.restoreAllMocks()
  })

  test("snapshot", () => {
    const { asFragment } = render(<Copyright />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("should be formatted as '© year author'", () => {
    const { getByText } = render(<Copyright />)
    expect(getByText("© 9999 author")).toBeInTheDocument()
  })
})
