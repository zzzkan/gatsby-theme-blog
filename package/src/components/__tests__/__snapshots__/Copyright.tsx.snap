// Vitest Snapshot v1

exports[`Copyright component > snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: sm;
}

<div
    class="chakra-text emotion-0"
  >
    Â© 9999 author
  </div>
</DocumentFragment>
`;
